test/cutting/test_cutting_roundtrip.py

```diff
@@ -179,28 +179,56 @@ def test_cutting_exact_reconstruction(example_circuit):
     assert np.allclose(exact_expvals, simulated_expvals, atol=1e-8)
 
 
-def test_sampler_with_identity_subobservable(example_circuit):
-    """This test ensures that the sampler does not throw an error if you pass it a subcircuit with no observable measurements.
+@pytest.mark.parametrize(
+    "sampler,is_exact_sampler", [(Sampler(), False), (ExactSampler(), True)]
+)
+def test_sampler_with_identity_subobservable(sampler, is_exact_sampler):
+    """This test ensures that the sampler works for a subcircuit with no observable measurements.
 
-    Tests temporary workaround to Issue #422.
+    Specifically, that
 
-    This test passes if no exceptions are raised.
+    - ``Sampler`` does not blow up (Issue #422); and
+    - ``ExactSampler`` returns correct results
 
+    This is related to https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/issues/422.
     """
+    # Create a circuit to cut
+    qc = QuantumCircuit(3)
+    append_random_unitary(qc, [0, 1])
+    append_random_unitary(qc, [2])
+    qc.rxx(np.pi / 3, 1, 2)
+    append_random_unitary(qc, [0, 1])
+    append_random_unitary(qc, [2])
 
-    qc = example_circuit
-    observable_to_test = PauliList(
+    # Determine expectation value using cutting
+    observables = PauliList(
         ["IIZ"]
     )  # Without the workaround to Issue #422, this observable causes a Sampler error.
     subcircuits, bases, subobservables = partition_problem(
-        qc, "AAB", observables=observable_to_test
+        qc, "AAB", observables=observables
     )
     subexperiments, coefficients = generate_cutting_experiments(
         subcircuits, subobservables, num_samples=np.inf
     )
-    samplers = {label: Sampler() for label in subexperiments.keys()}
+    samplers = {label: sampler for label in subexperiments.keys()}
     results = {
         label: sampler.run(subexperiments[label]).result()
         for label, sampler in samplers.items()
     }
-    _ = results
+    reconstructed_expvals = reconstruct_expectation_values(
+        results, coefficients, subobservables
+    )
+
+    if is_exact_sampler:
+        # Determine exact expectation values
```

Comment:
Apologies for this naive question but why use `ExactSampler()` and not just `Sampler()` here?

Reply:
As a result of the `pytest.mark.parametrize` decorator to this function, this test gets run twice, once with `Sampler()` and once with `ExactSampler()`.  The run using `Sampler()` has `is_exact_sampler is False`, so it's really just a smoke test: it makes sure the test runs without error, but it doesn't really test the result.  I made this choice because the Sampler is going to have shot noise, so the result it provides is only going to be approximate.  So, there is a second run with `ExactSampler()` with `is_exact_sampler is True`.  This "exact sampler" will return the exact probability distribution, without shot noise.  So we can inspect the reconstructed results and expect them to be exactly correct, up to roundoff error.

Reply:
I see, the part about shot noise addresses my question, thanks for the clarification.